src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php

```diff
@@ -339,13 +347,17 @@ public function iShouldNotSeeVariantWithOptionPricedAt(string $expectedOptionNam
 
         foreach ($variants as $variant) {
             foreach ($variant['optionValues'] as $optionValueIri) {
-                $optionValueData = $this->fetchItemByIri($optionValueIri);
-                $optionData = $this->fetchItemByIri($optionValueData['option']);
-
-                Assert::false(
-                    $optionData['name'] === $expectedOptionName &&
-                    $optionValueData['value'] === $expectedOptionValueValue,
-                );
+                $parts = explode('/', $optionValueIri);
+                $productOptionCode = $parts[5];
+                $productOptionValueCode = $parts[7];
+
+                if (StringInflector::nameToUppercaseCode($expectedOptionName) == StringInflector::nameToUppercaseCode($productOptionCode)) {
+                    Assert::notSame(
+                        StringInflector::nameToUppercaseCode($productOptionValueCode),
+                        StringInflector::nameToUppercaseCode($expectedOptionValueValue),
+                        sprintf('There is a variant with "%s" option value', $expectedOptionValueValue),
+                    );
+                }
```

Comment:
This check looks a bit "hacky", but as it is duplicated in two methods, at least, extract it to the separate private method